src/igvfd/types/file_set.py

```diff
@@ -317,6 +317,106 @@ def protocols(self, request, input_file_sets=None):
+            'type': 'string',
+            'notSubmittable': True,
+        }
+    )
+    def sample_summary(self, request, samples=None):
+        if not samples:
+            return
+
+        sample_terms = set()
+        sample_classifications = set()
+        treatments = set()
+        construct_library_set_types = set()
+        sorted = set()
+        targeted_genes_for_sorting = set()
+
+        treatment_purpose_to_adjective = {
+            'activation': 'activated',
+            'agonist': 'agonized',
+            'antagonist': 'antagonized',
+            'control': 'treated with a control',
+            'differentiation': 'differentiated',
+            'de-differentiation': 'de-differentiated',
+            'perturbation': 'perturbed',
+            'selection': 'selected',
+            'stimulation': 'stimulated'
+        }
+
+        for sample in samples:
+            sample_object = request.embed(sample, '@@object')
+
+            for term in sample_object['sample_terms']:
+                sample_term_object = request.embed(term, '@@object?skip_calculated=true')
+
+                if 'targeted_sample_term' in sample_object:
+                    targeted_sample_term_object = request.embed(
+                        sample_object['targeted_sample_term'], '@@object?skip_calculated=true')
+                    sample_terms.add(
+                        f"{sample_term_object['term_name']} induced to {targeted_sample_term_object['term_name']}")
+                else:
+                    sample_terms.add(sample_term_object['term_name'])
+            for classification in sample_object['classifications']:
+                sample_classifications.add(classification)
+
+            if 'construct_library_sets' in sample_object:
+                for construct_library_set in sample_object['construct_library_sets']:
+                    cls_object = request.embed(construct_library_set, '@@object?skip_calculated=true')
+                    construct_library_set_types.add(cls_object['file_set_type'])
+
+            if 'sorted_from' in sample_object:
+                sorted.add(True)
+                for file_set in sample_object['file_sets']:
+                    if file_set.startswith('/measurement-sets/'):
+                        fileset_object = request.embed(file_set, '@@object?skip_calculated=true')
+                        if 'targeted_genes' in fileset_object:
+                            for gene in fileset_object['targeted_genes']:
+                                gene_object = request.embed(gene, '@@object?skip_calculated=true')
+                                targeted_genes_for_sorting.add(gene_object['symbol'])
+
+            if 'treatments' in sample_object:
+                for treatment in sample_object['treatments']:
+                    treatment_object = request.embed(treatment, '@@object?skip_calculated=true')
+                    treatments.add(treatment_purpose_to_adjective[treatment_object['purpose']])
```

Comment:
This would error out if another treatment purpose is added to the enum list and the dicitionary treatment_purpose_to_adjective is not added, use .get() instead.

Reply:
Also can you rename this variable to treatment_purposes